app/scripts/components/exploration/components/datasets/colorRangeSlider.tsx

```diff
@@ -0,0 +1,272 @@
+import React, { useCallback, useEffect, useState, useRef } from 'react';
+
+import './color-range-slider.scss';
+import { colorMapScale } from '$components/exploration/types.d.ts';
+
+import { USWDSTextInput } from '$components/common/uswds';
+
+interface ColorrangeRangeSlideProps {
+  // Absolute minimum of color range
+  min: number;
+
+  // Absolute maximum of color range
+  max: number;
+
+  // Previously selected minimum and maximum of colorRangeScale
+  colorMapScale: colorMapScale | undefined;
+
+  // Update colorRangeScale
+  setColorMapScale: (colorMapScale: colorMapScale) => void;
+}
+
+export function ColorRangeSlider({
+  min,
+  max,
+  colorMapScale,
+  setColorMapScale
+}: ColorrangeRangeSlideProps) {
+  const setDefaultMin = colorMapScale?.min ? colorMapScale.min : min;
+  const setDefaultMax = colorMapScale?.max ? colorMapScale.max : max;
+
+  const [minVal, setMinVal] = useState(setDefaultMin);
+  const [maxVal, setMaxVal] = useState(setDefaultMax);
+  const minValRef = useRef<string | number>(setDefaultMin);
+  const maxValRef = useRef<string | number>(setDefaultMax);
+  const [inputError, setInputError] = useState({
+    min: false,
+    max: false,
+    largerThanMax: false,
+    lessThanMin: false
+  });
+
+  const range = useRef<HTMLDivElement>(null);
+
+  // Convert to percentage
+  const getPercent = useCallback(
+    (value) => ((value - min) * 100) / (max - min),
+    [min, max]
+  );
+  //Calculate the range
+  const rangeCalculation = (maxPercent, minPercent) => {
+    const thumbWidth = 20;
+    if (range.current) {
+      range.current.style.width = `calc(${
+        maxPercent - minPercent >= 100 ? 100 : maxPercent - minPercent
+      }% - ${(thumbWidth - minPercent * 0.2) * (maxPercent / 100)}px )`;
```

Comment:
There was a little bit of magic number math needed to get this working. Based off what I was able to locate in this link: https://css-tricks.com/value-bubbles-for-range-inputs/

Because the `left` and `width` are percentage we need to add a buffer of half the button width to both ends of the range. Doing so will ensure that the marker is centered when moving the left slider thumb. For the right slider thumb we needed to do the same but adjust the buffer value dynamically reduce the buffer on the right based on the max percent of the slider.